tests/integration/targets/ndo_node_setting/tasks/main.yml

```diff
@@ -0,0 +1,498 @@
+          - add_ns1_ptp.current.synce.qlOption == "op1"
+          - add_ns1_ptp.current.uuid is defined
+          - add_ns1_ptp.previous.description == ""
+          - add_ns1_ptp.previous.name == "ns1_new"
+          - add_ns1_ptp.previous.synce.adminState == "disabled"
+          - add_ns1_ptp.previous.synce.qlOption == "op1"
+          - add_ns1_ptp.previous.uuid is defined
+
+    - name: Add ns2 with synce and ptp values
+      cisco.mso.ndo_node_setting:
+        <<: *cm_update_ns1
+        name: ns2
+        state: "present"
+      register: add_ns2
+
+    - name: Assertion check for add ns2 with synce and ptp values
+      ansible.builtin.assert:
+        that:
+          - add_ns2 is changed
+          - add_ns2.current.description == "Updated description"
+          - add_ns2.current.name == "ns2"
+          - add_ns2.current.ptp.domain == 25
+          - add_ns2.current.ptp.prio1 == 128
+          - add_ns2.current.ptp.prio2 == 128
+          - add_ns2.current.synce.adminState == "enabled"
+          - add_ns2.current.synce.qlOption == "op2g1"
+          - add_ns2.current.uuid is defined
+          - add_ns2.previous == {}
+
+    - name: Create ns3 with empty synce
+      cisco.mso.ndo_node_setting:
+        <<: *mso_info
+        template: '{{ ansible_fabric_template | default("ansible_test") }}'
+        name: "ns3"
+        synce:
+          state: "enabled"
+        state: "present"
+      ignore_errors: true
+      register: add_ns3_synce
+
+    - name: Assertion check for create ns3 with empty synce
+      ansible.builtin.assert:
+        that:
+          - add_ns3_synce is changed
+          - add_ns3_synce.current.description == ""
+          - add_ns3_synce.current.name == "ns3"
+          - add_ns3_synce.current.uuid is defined
+          - add_ns3_synce.current.synce is not defined
+
+    - name: Create ns4 with empty ptp
+      cisco.mso.ndo_node_setting:
+        <<: *mso_info
+        template: '{{ ansible_fabric_template | default("ansible_test") }}'
+        name: "ns4"
+        ptp:
+          state: "enabled"
+        state: "present"
+      ignore_errors: true
+      register: add_ns4_ptp
+
+    - name: Assertion check for create ns4 with empty ptp
+      ansible.builtin.assert:
+        that:
+          - add_ns4_ptp is changed
+          - add_ns4_ptp.current.description == ""
+          - add_ns4_ptp.current.name == "ns4"
+          - add_ns4_ptp.current.uuid is defined
+          - add_ns4_ptp.current.ptp is not defined
+
+    - name: Query all Node Settings
+      cisco.mso.ndo_node_setting:
+        <<: *mso_info
+        template: '{{ ansible_fabric_template | default("ansible_test") }}'
+        state: "query"
+      register: query_all_ns_again
+
+    - name: Assertion check for query all Node Settings
+      ansible.builtin.assert:
+        that:
+          - query_all_ns_again is not changed
+          - query_all_ns_again.current | length == 4
```

Comment:
Should you also check that the 4 node settings are returned correctly? Maybe just checking each name is correct.